src/event.rs

```diff
@@ -19,6 +21,15 @@ pub enum Event {
         rw: ReadWrite,
         /// The address that triggered the fault.
         addr: *mut c_void,
+        /// The thread that triggered the fault, if [`FeatureFlags::THREAD_ID`] is enabled.
+        ///
+        /// If the thread ID feature is not enabled, the value of this field is undefined. It would
+        /// not be undefined behavior to use it, strictly speaking, but the [`Pid`] will not
+        /// necessarily point to a real thread.
```

Comment:
This, uh, sucks. But there's no way to look at the event in isolation and know whether this is a valid pid or not

Reply:
me: "what if we wrap this fd with some extra data to indicate later on if the pid should be valid, ourselves?"

Adam: turns out all we get during a fork event is the userfault fd and _not_ flags, so this is inconsistent with using and supporting uffd across fork events.

this really is the best option for the time being :grimacing: